libs/common-services/src/main/java/uk/gov/di/ipv/core/library/service/CiMitUtilityService.java

```
@@ -38,12 +43,29 @@ public Optional<JourneyResponse> getCiMitigationJourneyStep(ContraIndicators con
         var cimitConfig = configService.getCimitConfig();
         for (var ci : contraIndicators.getContraIndicatorsMap().values()) {
             if (isCiMitigatable(ci) && !isBreachingCiThresholdIfMitigated(ci, contraIndicators)) {
-                return Optional.of(new JourneyResponse(cimitConfig.get(ci.getCode())));
+                MitigationRoute mitigationRoute =
+                        getMitigationRoute(cimitConfig.get(ci.getCode()), ci.getDocument());
+                return mitigationRoute == null
+                        ? Optional.empty()
```

I'm wondering if we should throw here, rather than returning an empty optional. We've already checked that the CI can be mitigated, so we _should_ be able to find a route. If this was null I think it would mean that we've somehow not matched the document ID on the CI to a document in the config, which I don't think should happen. Rather than using `orElseGet` in the stream in `getMitigationRoute` could use `orElseThrow`?

---

Earlier I thought of throwing exception but then thought logging it that config is missing. If we throw exception then at handler level we need to catch it and return to /journey/error route.

---

I think we want to return /journey/error though, as somethign wrong has happened. If we just log it we won't become aware of it?

---

done